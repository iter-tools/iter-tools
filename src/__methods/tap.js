/**
 * @generated-from ./../impls/$tap/tap.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

export { __tap as default } from '../impls/$tap/tap.js';
