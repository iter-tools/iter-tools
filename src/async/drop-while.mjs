import asyncIter from './async-iter'

export default function dropWhile (func, iterable) {
  async function * curriedDropWhile (i) {
    let drop = true
    let c = 0
    for await (const item of asyncIter(i)) {
      if (!drop) {
        yield item
      } else {
        drop = func(item, c++)
        if (!drop) {
          yield item
        }
      }
    }
  }
  if (iterable) {
    return curriedDropWhile(iterable)
  }
  return curriedDropWhile
}
