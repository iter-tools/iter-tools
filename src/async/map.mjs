import asyncIter from './async-iter'

export default function map (func, iterable) {
  async function * curriedMap (i) {
    let c = 0
    for await (const item of asyncIter(i)) {
      yield func(item, c++)
    }
  }
  if (iterable) {
    return curriedMap(iterable)
  }
  return curriedMap
}
