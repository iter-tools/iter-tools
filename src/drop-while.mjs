import iter from './iter'

function * dropWhile (func, iterable) {
  let drop = true
  let c = 0
  for (const item of iter(iterable)) {
    if (!drop) {
      yield item
    } else {
      drop = func(item, c++)
      if (!drop) {
        yield item
      }
    }
  }
}

export default function curriedDropWhile (func, iterable) {
  if (!iterable) {
    return iterable => dropWhile(func, iterable)
  }
  return dropWhile(func, iterable)
}
