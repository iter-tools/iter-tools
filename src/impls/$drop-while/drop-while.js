/**
 * @generated-from ./$drop-while.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

import { iterableCurry } from '../../internal/iterable.js';

export function* dropWhile(source, predicate) {
  let drop = true;
  let c = 0;
  for (const value of source) {
    if (!drop) {
      yield value;
    } else {
      drop = predicate(value, c++);
      if (!drop) {
        yield value;
      }
    }
  }
}

export default /*#__PURE__*/ iterableCurry(dropWhile);
