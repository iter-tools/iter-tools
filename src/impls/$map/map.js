/**
 * @generated-from ./$map.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

import { iterableCurry } from '../../internal/iterable.js';

export function* __map(source, func) {
  let c = 0;
  for (const value of source) {
    yield func(value, c++);
  }
}

export const map = /*#__PURE__*/ iterableCurry(__map);
