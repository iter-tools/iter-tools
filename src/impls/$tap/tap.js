/**
 * @generated-from ./$tap.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

import { iterableCurry } from '../../internal/iterable.js';

export function* tap(source, callback) {
  let c = 0;
  for (const value of source) {
    callback(value, c++);
    yield value;
  }
}

export default /*#__PURE__*/ iterableCurry(tap);
