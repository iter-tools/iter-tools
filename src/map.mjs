import iter from './iter'

function * map (func, iterable) {
  let c = 0
  for (const item of iter(iterable)) {
    yield func(item, c++)
  }
}

export default function curriedMap (func, iterable) {
  if (!iterable) {
    return iterable => map(func, iterable)
  }
  return map(func, iterable)
}
