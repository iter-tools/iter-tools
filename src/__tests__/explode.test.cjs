const { default: pluginTester } = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

// For now this must be run manually. To run it you must first do
// ln -s ../ node_modules/iter-tools-es
// It should be possible to fix this once resolve supports self references
// (and a version that does is integrated into babel-plugin-macros).
// https://github.com/browserify/resolve/pull/224

// eslint-disable-next-line jest/no-disabled-tests
describe.skip('explode.macro', () => {
  pluginTester({
    plugin,
    title: 'explode.macro',
    snapshot: true,
    babelOptions: {
      filename: __filename,
    },
    tests: [
      `
      import 'above';
      import { map as snap, dropWhile } from 'iter-tools-es/src/explode.macro.cjs';
      import { tap, __tap } from 'iter-tools-es/src/explode.macro.cjs';
      import 'below';
    `,
    ],
  });
});

// I can't keep the snapshot for a disabled test, so when turning it back on make
// sure the output is as expected:
/*
import 'above';
import { map as snap } from 'iter-tools-es/methods/map';
import { dropWhile } from 'iter-tools-es/methods/drop-while';
import { tap } from 'iter-tools-es/methods/tap';
import { __tap } from 'iter-tools-es/__methods/tap';
import 'below';
*/
