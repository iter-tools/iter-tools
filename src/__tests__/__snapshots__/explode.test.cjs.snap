// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. unknown plugin: 1. unknown plugin 1`] = `

import 'above';
import { map as snap, dropWhile } from '../explode.macro.cjs';
import { tap } from '../explode.macro.cjs';
import 'below';

      ↓ ↓ ↓ ↓ ↓ ↓

import 'above';
import { map as snap } from 'iter-tools/methods/map';
import { dropWhile } from 'iter-tools/methods/drop-while';
import { tap } from 'iter-tools/methods/tap';
import 'below';


`;
